app/src/main/java/net/gsantner/opoc/ui/FilesystemViewerAdapter.java

```diff
@@ -239,6 +252,14 @@ public boolean isCurrentFolderVirtual() {
         );
     }
 
+    @Override
+    public void onImageLoaded(Bitmap bitmap, ilesystemViewerViewHolder holder) {
```

Comment:
After removing previous code from this function, the image file shows a grey tinted icon instead of the thumbnail.

<img src="https://user-images.githubusercontent.com/55661063/66520415-a72b9e80-ead8-11e9-93cf-ac073eebcbb0.png" width="400" alt="Image File">

I think this and Issue 2 is related to updation inside onBindViewHolder and RecyclerView re-using old views.

Reply:
@gsantner Just a guess but I think updation in onBindViewHolder and the fact that RecyclerView re-use old views have something to do with this and issue 2.

Reply:
Would need todo it myself to fix all issues, but guess you'll find a way :)